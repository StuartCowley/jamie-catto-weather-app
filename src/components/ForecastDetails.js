import React from 'react';
import '../styles/ForecastDetails.css';
import PropTypes from 'prop-types';
import WeatherIcon from 'react-icons-weather';

function ForecastDetails({forecast}) {
    const {
        date, temperature, description, icon,
        wind, humidity
      } = forecast;
    const formattedDate = new Date(date).toDateString();
    return (
        <div className="forecast-details">
        <div className="forecast-details__date">
            {formattedDate}
        </div>
        <div className="forecast-details__temperature-min">
            Minimum: {temperature.min}
            &deg;C
        </div>
        <div className="forecast-details__temperature-max">
            Maximum: {temperature.max}
            &deg;C
        </div>
        <div className="forecast-details__wind">
            Wind: {wind.speed} 
            {wind.direction}
        </div>
        <div className="forecast-details__humidity">
            Humidity: {humidity} 
        </div>
        <div className="forecast-details__description">
            {description}
        </div>
        <div className="forecast-details__icon" data-testid="forecast-icon">
            <WeatherIcon name="owm" iconId={icon}/>
        </div>
        </div>
    )
}

ForecastDetails.propTypes = {
    forecast: PropTypes.shape({
        date: PropTypes.number,
        temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
        }),
        description: PropTypes.string,
        icon: PropTypes.string,
        wind: PropTypes.shape({
            speed: PropTypes.number,
            direction: PropTypes.string,
        }),
        humidity: PropTypes.number,
    }).isRequired,
}

export default ForecastDetails;