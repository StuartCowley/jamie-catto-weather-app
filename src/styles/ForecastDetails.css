.forecast-details {
    margin-top: 20px;
    width: 10%;
    padding-left: 45%;
}

.forecast-details > div {
    margin-top: 3px;
}