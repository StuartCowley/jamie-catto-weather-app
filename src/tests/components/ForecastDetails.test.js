import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastSummary", () => {
    const propForecast = {
      date: 1111111,
      description: "Stub description",
      icon: "800",
      temperature: {
        min: 12,
        max: 22,
      },
      wind: {
        speed: 15,
        direction: 's',
      },
    humidity: 30,
    };
  
    it("renders correctly", () => {
      const { asFragment } = render(
        <ForecastDetails
          forecast={propForecast}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    });
});