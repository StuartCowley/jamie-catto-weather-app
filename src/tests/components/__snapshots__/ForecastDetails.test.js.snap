// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature-min"
    >
      Minimum: 12°C
    </div>
    <div
      class="forecast-details__temperature-max"
    >
      Maximum: 22°C
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind: 15s
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30
    </div>
    <div
      class="forecast-details__description"
    >
      Stub description
    </div>
    <div
      class="forecast-details__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
  </div>
</DocumentFragment>
`;
